File: gbs-pos-api/internal/model/cart_display.go
package model

import (
	"time"

	"gorm.io/datatypes"
)

// CartDisplay stores the latest JSON payload for a POS terminal customer display.
// The backend treats Payload as an opaque JSONB document.
type CartDisplay struct {
	ID         uint           `gorm:"primaryKey"`
	TerminalID string         `gorm:"uniqueIndex;size:64;not null"`
	Payload    datatypes.JSON `gorm:"type:jsonb;not null"`

	// Device information columns (nullable for backward compatibility)
	DeviceModel        *string `gorm:"size:100"`
	DeviceManufacturer *string `gorm:"size:100"`
	DeviceBrand        *string `gorm:"size:100"`
	AndroidVersion     *string `gorm:"size:20"`
	SDKInt             *int    `gorm:"type:integer"`
	AppVersion         *string `gorm:"size:50"`
	AppVersionCode     *int64  `gorm:"type:bigint"`

	UpdatedAt time.Time
}

File: cms-web/src/api/cartDisplay.ts
import apiClient from './client'
import type { ApiResponse } from '../types/api'

export interface DeviceInfo {
  deviceModel: string | null
  deviceManufacturer: string | null
  deviceBrand: string | null
  androidVersion: string | null
  sdkInt: number | null
  appVersion: string | null
  appVersionCode: number | null
}

export interface Terminal {
  terminalId: string
  payload: string
  deviceModel: string | null
  deviceManufacturer: string | null
  deviceBrand: string | null
  androidVersion: string | null
  sdkInt: number | null
  appVersion: string | null
  appVersionCode: number | null
  updatedAt: string
}

export async function getTerminals(): Promise<ApiResponse<Terminal[]>> {
  const response = await apiClient.get<ApiResponse<Terminal[]>>('/v1/display/terminals')
  return response.data
}
